evm/utils/db.py

```diff
@@ -3,6 +3,8 @@ def make_block_hash_to_number_lookup_key(block_hash):
     return hash_to_number_key
 
 
+# XXX: Isn't this going to be a problem when we have multiple chains, since
+# then there can be multiple blocks with a given number?
```

Comment:
Yes, this whole scheme is going to need to be reworked to deal with competing chain histories.